qiling/loader/elf.py

```diff
@@ -147,88 +147,86 @@ def seg_perm_to_uc_prot(perm: int) -> int:
         return prot
 
     def load_with_ld(self, elffile: ELFFile, stack_addr: int, load_address: int, argv: Sequence[str] = [], env: Mapping[str, str] = {}):
-        # get list of loadable segments; these segments will be loaded to memory
-        seg_pt_load = tuple(seg for seg in elffile.iter_segments() if seg['p_type'] == 'PT_LOAD')
-
-        # determine the memory regions that need to be mapped in order to load the segments.
-        # note that region boundaries are aligned to page, which means they may be larger than
-        # the segment they contain. to reduce mapping clutter, adjacent regions with the same
-        # perms are consolidated into one contigous memory region
-        load_regions: Sequence[Tuple[int, int, int]] = []
-
-        # iterate over loadable segments by vaddr
-        for seg in sorted(seg_pt_load, key=lambda s: s['p_vaddr']):
-            lbound = self.ql.mem.align(load_address + seg['p_vaddr'])
-            ubound = self.ql.mem.align_up(load_address + seg['p_vaddr'] + seg['p_memsz'])
-            perms = QlLoaderELF.seg_perm_to_uc_prot(seg['p_flags'])
-
-            if load_regions:
-                prev_lbound, prev_ubound, prev_perms = load_regions[-1]
-
-                # new region starts where the previous one ended
-                if lbound == prev_ubound:
-                    # same perms? extend previous memory region
-                    if perms == prev_perms:
-                        load_regions[-1] = (prev_lbound, ubound, prev_perms)
-
-                    # different perms? start a new one
-                    else:
+
+        def load_elf_segments(self, elffile: ELFFile, load_address: int, info: str):
```

Comment:
This is an internal function, not an instance method, so no need to pass `self` as an arg.
The function can use `self` from the outter scope if needed.